server/s3_motr_reader.cc

```diff
@@ -344,6 +344,59 @@ size_t S3MotrReader::get_next_block(char **data) {
   *data = (char *)motr_rw_op_context->data->ov_buf[iteration_index];
   data_read = motr_rw_op_context->data->ov_vec.v_count[iteration_index];
   iteration_index++;
+
+  size_t length = data_read;
+  s3_log(S3_LOG_DEBUG, "", "%s total_size_to_read=%zu total_size_read=%zu length=%zu", __func__, total_size_to_read, total_size_read, length);
+  if (total_size_to_read < total_size_read + length) {
+    s3_log(S3_LOG_DEBUG, "", "%s length changed before %zu: ", __func__, length);
+    length = total_size_to_read - total_size_read;
+    s3_log(S3_LOG_DEBUG, "", "%s length changed after %zu: ", __func__, length);
+  }
+  if (multipart_part_size == 0) {
+    // non-multipart-upload case. Just calculate md5 of the entire object.
+    md5crypt.Update(*data, length);
+  } else {
+    // the object was created with multipart upload.
+    // Calculate md5 checksums for each part independently.
+    size_t begin = total_size_read;
```

Comment:
Not able to understand below logic, can you please put more comment here

Reply:
Re-implemented the chunk of code to make it easier to read, but haven't re-tested it. Will post another comment once this chunk of code is ready for review.